src/check.h

```diff
@@ -188,4 +188,11 @@ extern Expr *statMpz;
 extern Expr *statMpq;
 extern Expr *statType;
 
+/**
+ * Given a type, `e`, computes its kind.
+ *
+ * In particular, this will be `statType` if the kind is that of proper types.
```

Comment:
Can you give a bit more examples?  What other return values can be given here?  E.g. I assume nullptr = not proper type?  Also, summarize what it means to have proper type.

Reply:
A fair critique. I've written more about this now.